### PR TITLE
Neptune fix bug init run

### PR DESCRIPTION
**Pull request by AleksanderWWW**
# What does this PR do?

We realised that the `init_run` function embedded in the integration was accepting a deprecated kwarg `run` which was replaced with `with_id` some time ago. Without this fix there might be cases where the NeptuneCallback will not run correctly and throw an error, that the function received an unexpected argument.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by kshitij12345 — COMMENTED** [review]
We should update these imports so that we don't raise Deprecation warnings for the latest client

https://github.com/huggingface/transformers/blob/b1ea0b9ab8da4152c9aff7be8791e1c14a8c0c54/src/transformers/integrations.py#L1223-L1224

File: src/transformers/integrations.py [hunk 1]
```diff
@@ -1245,7 +1249,7 @@ def _ensure_run_with_monitoring(self):
                 return
 
             if self._run and not self._is_monitoring_run and not self._force_reset_monitoring_run:
-                self._initialize_run(run=self._run_id)
+                self._initialize_run(with_id=self._run_id)
```
[the remark this answers is not in the document]
Comment (amyeroberts) [reply src/transformers/integrations.py:1252]: @AleksanderWWW Yep, the failing tests came from a new release of optax. We're resolved this on `main`.

**Review by sgugger — APPROVED** [review]
Thanks for the fix.

**Comment by AleksanderWWW** [comment]
@sgugger Do you know more or less when will that be released?

**Comment by amyeroberts** [comment]
@AleksanderWWW It was released this week in v4.29.0

**Comment by AleksanderWWW** [comment]
Ah yes, my bad. I didn't realize that I had a bug in my own tests :smile: Thank you @amyeroberts!

**Pull request merged**